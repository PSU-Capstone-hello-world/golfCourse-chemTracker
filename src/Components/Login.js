import React, { useState } from 'react';
import {Container, Row, Button, Form} from 'react-bootstrap';
import PropTypes from 'prop-types';
import '../Styles/Login.css';

async function loginUser(credentials) {
return fetch('http://localhost:8080/login', {
    method: 'POST',
    headers: {
    'Content-Type': 'application/json'
    },
    body: JSON.stringify(credentials)
})
    .then(data => data.json())
}


Login.propTypes = {
    setToken: PropTypes.func.isRequired
}


function Login() {
    const [validated, setValidated] = useState(false);
    const [username, setUserName] = useState();
    const [password, setPassword] = useState();
    const [token, setToken] = useState();

    const handleSubmit = async event => {
        const form = event.currentTarget;

        if (form.checkValidity() === false) {
            event.preventDefault();
            event.stopPropagation();
        }

        setValidated(true);
        const token = await loginUser({
            username,
            password
          });

        setToken(token);
    }

        if(!token) {
        return (
            <Container id='containerLogin'>
                <Row className='justify-content-center align-self-center'>
                    <Form id='login' validated={validated} onSubmit={handleSubmit}>
                        <div className='test d-flex justify-content-center'>
                            <h3>Sign In</h3>
                        </div>

                        <Form.Group controlId='formEmail'>
                            <Form.Label>Email Address</Form.Label>
                            <Form.Control type='email' required onChange={e => setUserName(e.target.value)} placeholder='Enter email' />
                        </Form.Group>

                        <Form.Group controlId='formPassword'>
                            <Form.Label>Password</Form.Label>
                            <Form.Control type='password' required placeholder='Password' onChange={e => setPassword(e.target.value)}/>
                        </Form.Group>

                        {/* will need to change the url for this once we have authentication up and going */}
                        <Button type='submit' variant='info' className='btn-block'>Submit</Button>
                    </Form>
                </Row>
            </Container>
        )
        }
}

export default Login;