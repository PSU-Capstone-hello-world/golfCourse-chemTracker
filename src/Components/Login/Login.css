#containerLogin {
    max-width: 1000px;
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
}

#loginForm {
    border: 1px solid black;
    width: 90%;
    margin: 0 auto;
    padding: 20px 20px;
    border-radius: 20px;
    background-color: white;
}
