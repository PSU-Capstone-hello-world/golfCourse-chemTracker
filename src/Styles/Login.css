#containerLogin {
    max-width: 1000px;
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
}

#loginForm {
    border: 1px solid black;
    width: 90%;
    margin: 0 auto;
    padding: 20px 20px;
    border-radius: 5px;
    background: rgb(207,203,203);
    background: linear-gradient(90deg, rgba(207,203,203,0.78) 0%, rgba(255,255,255,0.819765406162465) 35%, rgba(207,203,203,0.7777485994397759) 100%);
}
