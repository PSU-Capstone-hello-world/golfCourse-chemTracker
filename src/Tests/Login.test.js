import React from 'react';
import testRenderer from 'react-test-renderer';
import { shallow, configure, mount } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import Login from '../Components/Login';

configure({ adapter: new Adapter() });
const mockToken = jest.fn();

describe('Login component', function() {
    it('component renders correctly', () => {
        shallow(<Login setToken={mockToken} />);
        const tree = testRenderer.create(<Login setToken={ mockToken }/>).toJSON();
        expect(tree).toMatchSnapshot();
    })
})