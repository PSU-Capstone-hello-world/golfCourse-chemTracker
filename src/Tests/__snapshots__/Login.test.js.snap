// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component component renders correctly 1`] = `
<div
  className="container"
  id="containerLogin"
>
  <div
    className="justify-content-center align-self-center row"
  >
    <form
      className=""
      id="loginForm"
      onSubmit={[Function]}
      preventDefault={true}
    >
      <div
        className="test d-flex justify-content-center"
      >
        <h3>
          Log in
        </h3>
      </div>
      <div
        className="form-group"
      >
        <label
          className="form-label"
          htmlFor="formEmail"
        >
          Email Address
        </label>
        <input
          className="form-control"
          id="formEmail"
          onChange={[Function]}
          placeholder="Enter email"
          required={true}
          type="email"
        />
      </div>
      <div
        className="form-group"
      >
        <label
          className="form-label"
          htmlFor="formPassword"
        >
          Password
        </label>
        <input
          className="form-control"
          id="formPassword"
          onChange={[Function]}
          placeholder="Password"
          required={true}
          type="password"
        />
      </div>
      <button
        className="btn-block btn btn-info"
        disabled={false}
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>
</div>
`;
